networkit/cpp/centrality/LocalSquareClusteringCoefficient.cpp

```diff
@@ -0,0 +1,53 @@
+// no-networkit-format
+#include <networkit/centrality/LocalSquareClusteringCoefficient.hpp>
+#include <networkit/linkprediction/NeighborhoodUtility.hpp>
+#include <networkit/graph/Graph.hpp>
+#include <omp.h>
+
+namespace NetworKit {
+
+LocalSquareClusteringCoefficient::LocalSquareClusteringCoefficient(const Graph& G) : Centrality(G, false, false) {
+    if (G.isDirected()) throw std::runtime_error("Not implemented: Local square clustering coefficient is currently not implemented for directed graphs");
+    if (G.numberOfSelfLoops()) throw std::runtime_error("Local square clustering coefficient implementation does not support graphs with self-loops. Call Graph.removeSelfLoops() first.");
+}
+
+void LocalSquareClusteringCoefficient::run() {
+    count z = G.upperNodeIdBound();
+    scoreData.clear();
+    scoreData.resize(z);
+
+    G.balancedParallelForNodes([&](const node u) {
+        double squares = 0;  // Number of squares found.
+        double potential_squares = 0;  // Maximum number of possible squares.
+        const auto neighborsU = G.neighborRange(u);
+
+        // Iterate over all combinations of neighbors.
+        for (auto iterV = neighborsU.begin(); iterV != neighborsU.end(); iterV++) {
+            const auto neighborsV = G.neighborRange(*iterV);
+            auto iterW = iterV;
+            for (++iterW; iterW != neighborsU.end(); iterW++) {
```

Comment:
This is a bit simpler:
```suggestion
            for (auto iterW = std::next(iterV); iterW != neighborsU.end(); ++iterW) {
```